### PR TITLE
ISS: Collect garbage state

### PR DESCRIPTION
**Pull request by sergefdrv**
This PR aims at resolving #10

### CODE REVIEW

**Review by matejpavlovic — APPROVED** [review]
Beautiful 👍 :)

File: pkg/iss/iss.go [hunk 1]
```diff
@@ -513,9 +509,12 @@ func (iss *ISS) applyRequestReady(requestReady *eventpb.RequestReady) *events.Ev
 }
 
 // applyAppSnapshot applies the event of the application creating a state snapshot.
-// It passes the snapshot to the appropriate CheckpointTracker (identified by the event's associated sequence number).
+// It passes the snapshot to the appropriate CheckpointTracker (identified by the event's associated epoch number).
 func (iss *ISS) applyAppSnapshot(snapshot *eventpb.AppSnapshot) *events.EventList {
-	return iss.getCheckpointTracker(t.SeqNr(snapshot.Sn)).ProcessAppSnapshot(snapshot.Data)
+	if iss.epoch.Nr != t.EpochNr(snapshot.Epoch) {
```
[the remark this answers is not in the document]
Comment (sergefdrv) [reply pkg/iss/iss.go:510]: Not sure if we really need to handle late snapshots, but we certainly need to employ some form of flow control between application and protocol module as described in #57

File: pkg/iss/iss.go [hunk 2]
```diff
@@ -160,23 +160,19 @@ type ISS struct {
...
+	epochs map[t.EpochNr]*epochInfo
 
 	// Index of orderers based on the buckets they are assigned.
 	// For each bucket ID, this map stores the orderer to which the bucket is assigned in the current epoch.
 	bucketOrderers map[int]sbInstance
 
 	// --------------------------------------------------------------------------------
-	// These fields are modified throughout an epoch
+	// These fields are modified throughout an epoch.
+	// TODO: Move them into `epochInfo`?
```
[the remark this answers is not in the document]
Comment (sergefdrv) [reply pkg/iss/iss.go:175]: To be resolved in #62

File: pkg/iss/iss.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (matejpavlovic) [reply pkg/iss/iss.go:510]: Yes. But for the sake of this PR, I think we're good.

**Pull request merged**